packages/backend/src/server/routers/campaign.ts

```diff
@@ -28,7 +32,8 @@ export const campaignRouter = router({
         throw new Error('MissingLogin')
       } else if (!isKyc(ctx.session.user)) {
         throw new Error('MissingKYC')
-      } else {
+      }
+      if (input.address) {
```

Comment:
address is not optional, don't ignore it if it is not provided.
Also this path is not validated, the user could specify whatever contract address he likes, there is no check that the contract was deployed from the campaign factory nor that it was deployed from this user.